src/proof_rules/mod.rs

```diff
@@ -140,19 +142,25 @@ struct ProcContext {
 
 /// Walks the AST and transforms encoding annotations into their desugared form.
 /// Correct and sound usage of the encoding annotations are also checked during this process.
-pub struct EncodingVisitor<'tcx, 'sunit> {
+pub struct EncodingVisitor<'tcx, 'sunit, 'hmap> {
     tcx: &'tcx mut TyCtx,
     source_units_buf: &'sunit mut Vec<Item<SourceUnit>>,
+    looping_procs: &'hmap HashMap<Ident, Span>, // The set of looping procedures along with the spans of their looping calls
```

Comment:
Is there a reason why the `looping_procs` is owned by some other object? Seems not necessary and complicated?

Reply:
Also, you can make the comment a proper Rustdoc comment on this field.